### PR TITLE
Track previous catkeys

### LINKED ISSUE #3057: Move set previous catkey logic to argo
There is currently logic in the [catkey convenience method](https://github.com/sul-dlss/dor-services/blob/c2c7eeb84cfae6e84a7b636f2e08159838cfe699/lib/dor/datastreams/identity_metadata_ds.rb#L124-L140) that adds a previous catkey when a new catkey is set.

This logic should be applied when a catkey is changed.

Currently this logic is applied at the datastream class level in dor-services gem.  The logic needs to be moved to argo at the location where we update catkeys and needs to apply the change to the cocina model.

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made?

Fixes #3057

### CODE REVIEW

File: app/models/concerns/catkey.rb [hunk 1]
```diff
@@ -1,11 +1,50 @@
 # frozen_string_literal: true
 
-module Catkey
+class Catkey
```
Comment (mjgiarlo) [inline app/models/concerns/catkey.rb:3]: Should this move out of `app/models/concerns/` now that it's a class?

**Review by mjgiarlo — APPROVED** [review]
Could consider moving `Catkey` but otherwise LGTM. Approving.

File: app/models/concerns/catkey.rb (hunk 1 above)
Comment (jcoyne) [reply app/models/concerns/catkey.rb]: Good idea. I moved it.

File: app/services/catkey.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class Catkey
+  SYMPHONY = 'symphony'
+  PREVIOUS_CATKEY = 'previous_catkey'
```
Comment (justinlittman) [inline app/services/catkey.rb:5]: This should be `previous symphony`.

File: app/services/catkey.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  def initialize(model)
+    @model = model
+  end
+
+  def deserialize
+    symphony_links.join(', ')
+  end
+
+  # If there was already a catkey in the record, store that in the "previous" spot (assuming there is no change)
+  # @return [Array<Hash>] a list of catalog links
```
Comment (justinlittman) [inline app/services/catkey.rb:24]: Can you document the supported format in Rdoc?

File: app/services/catkey.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  def serialize(catkey)
+    new_catkeys = split(catkey)
+    removed_links = symphony_links - new_catkeys
+    links = (previous_links + removed_links).map { |record_id| { catalog: 'previous_catkey', catalogRecordId: record_id } }
```
Comment (justinlittman) [inline app/services/catkey.rb:28]: This should be `previous symphony`.

**Review by justinlittman — CHANGES_REQUESTED** [review]

File: app/services/catkey.rb (hunk 3 above)
Comment (jcoyne) [reply app/services/catkey.rb:24]: @justinlittman the format of the catkey parameter?

File: app/services/catkey.rb (hunk 4 above)
Comment (jcoyne) [reply app/services/catkey.rb:28]: @justinlittman isn't that the behavior of the current system?

**Review by justinlittman — APPROVED** [review]

**Pull request merged**